### PR TITLE
Added the neostore (neo4j db) for noe4jdb dataset 

### PR DESCRIPTION
**Pull request by zipho**
The neostore is a composite datatype for noe4jdb dataset. 

The neostore datatype is currently a dependent to the neo4j interactive environment.

A pull request for the GIE: https://github.com/bxlab/galaxy_hackathon/pull/34

A tool that utilizes this composite datatype: https://testtoolshed.g2.bx.psu.edu/view/sanbi-uwc/build_ctb_gene/74ad74a4be56

### CODE REVIEW

File: lib/galaxy/datatypes/neo4j.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+verbose = True
+   
+   
+class Neo4j(Html):
+    """ 
+    base class to use for neostore datatypes
+    derived from html - composite datatype elements
+    stored in extra files path
+    """
+    #def __init__(self):
```
Comment (nturaga) [inline lib/galaxy/datatypes/neo4j.py]: I guess these commented lines are not needed. @zipho

Comment (zipho) [inline lib/galaxy/datatypes/neo4j.py]: Thanks @nturaga .. removed those.

**Comment by nturaga** [comment]
@zipho It'd be super if you can provide a small test case for this data type under `galaxy/test-data` and add it to your PR.

**Comment by zipho** [comment]
@nturaga  The datatype is the dependency of neo4j GIE https://github.com/galaxyproject/galaxy/pull/2610 and dataset of this type will be produced by the build_ctb_gene tool. https://testtoolshed.g2.bx.psu.edu/view/sanbi-uwc/build_ctb_gene/74ad74a4be56

I assume by test-case, you mean i should add sample dataset of this datatype? However I just realized I could disable the upload for this datatype for now, as it's not currently a requirement for the tool.

**Comment by nturaga** [comment]
@zipho Yes, i meant add a sample dataset of this datatype to `galaxy/test-data`.

**Comment by zipho** [comment]
@nturaga Added a neostore.zip sample datatype into `galaxy/test-data` .  Upload for the datatype is set to False.

**Comment by jmchilton** [comment]
@galaxybot test this

**Comment by nturaga** [comment]
@galaxybot test this

**Comment by jmchilton** [comment]
Is neostore.zip as small as can be? It will add almost a Mb to the repo size - I'm okay with this but if it can be smaller that would be better.

**Comment by zipho** [comment]
@jmchilton I managed to reduce the size of neostore.zip to 8.0 K

**Comment by zipho** [comment]
Not sure why this fails all of sudden as I only added the noestore zip file.

**Comment by mvdbeek** [comment]
The test-failures may coincide with a new flake8 version that has been uploaded to pypi today.
I'm having the same problem in #2554

**Comment by jmchilton** [comment]
Awesome sauce - thanks for working with us on this! I'm very excited to see exciting neo4j applications on Galaxy.

**Pull request merged**